### ISSUE TITLE
cmd/compile: implement global new(T) without calling newobject

### ISSUE THREAD

**Issue by josharian** [golang/go#29691]
```go
package p

var x = new(int)
```

This generates init code to call `newobject` (including a write barrier, which I thought wasn't supposed to happen any more?). Instead, we should make a BSS int symbol, and then make `x` contain a relocation to that symbol.

cc @randall77 @mvdan

**Comment by randall77** [comment]
No, we still need a write barrier when initializing a global location.

The write barriers I changed recently removes them when initializing the contents of the results of a `newobject` call. Not sure if you're referring to that.

Yes, we can and should allocate the `int` statically.  Small caveat: watch out for #29068.

**Comment by Skarlso** [comment]
Hi @josharian! Can I help?

**Comment by josharian** [comment]
Go for it. You'll be looking at sinit.go. It's not particularly clear code, even by compiler standards. I recommend writing yourself a few simplified examples to work off of.

**Comment by Skarlso** [comment]
Sounds lovely. :D But at least I'll learn while trying to figure it out. Which is always good. :)

**Comment by Skarlso** [comment]
Picking this up now. Sorry, was pre-occupied with another issue on packages. :)
